src/logger/src/logger.rs

```diff
@@ -1,776 +1,343 @@
-        // does not attempt to flush.
-        if writeln!(writer, "{msg}")
-            .and_then(|_| writer.flush())
-            .is_err()
-        {
+            // If the write returns an error, increment missed log count.
             // No reason to log the error to stderr here, just increment the metric.
-            METRICS.logger.missed_log_count.inc();
+            if result.is_err() {
+                METRICS.logger.missed_log_count.inc();
+            }
         }
     }
-}
-
-/// Describes the errors which may occur while handling logging scenarios.
-#[derive(Debug, thiserror::Error)]
-pub enum LoggerError {
-    /// Initialization Error.
-    #[error("Logger initialization failure: {0}")]
-    Init(init::Error),
-}
-
-/// Implements the "Log" trait from the externally used "log" crate.
-impl Log for Logger {
-    fn enabled(&self, _metadata: &Metadata) -> bool {
-        // No filtering beyond what the log crate already does based on level.
-        true
-    }
-
-    fn log(&self, record: &Record) {
-        let msg = format!(
-            "{} {} {}",
-            LocalTime::now(),
-            self.create_prefix(record),
-            record.args()
-        );
-        self.write_log(&msg);
-    }
 
     // This is currently not used.
     fn flush(&self) {
         unreachable!();
     }
 }
 
-#[cfg(test)]
-mod tests {
-    use std::fs::{create_dir, read_to_string, remove_dir, remove_file, OpenOptions};
-    use std::io::{BufWriter, Read, Write};
-    use std::sync::Arc;
-
-    use log::{info, Level};
-
-    use super::*;
-
-    const TEST_INSTANCE_ID: &str = "TEST-INSTANCE-ID";
-    const TEST_APP_HEADER: &str = "App header";
-
-    const LOG_SOURCE: &str = "logger.rs";
-    const LOG_LINE: u32 = 0;
-
-    struct LogWriter {
-        buf: Arc<Mutex<Vec<u8>>>,
-    }
-
-    impl Write for LogWriter {
-        fn write(&mut self, buf: &[u8]) -> std::io::Result<usize> {
-            let mut data = self.buf.lock().unwrap();
-            data.append(&mut buf.to_vec());
-
-            Ok(buf.len())
-        }
+/// Strongly typed structure used to describe the logger.
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
+#[serde(deny_unknown_fields)]
+pub struct LoggerConfig {
+    /// Named pipe or file used as output for logs.
+    pub log_path: Option<PathBuf>,
+    /// The level of the Logger.
+    pub level: Option<LevelFilter>,
+    /// Whether to show the log level in the log.
+    pub show_level: Option<bool>,
+    /// Whether to show the log origin in the log.
+    pub show_log_origin: Option<bool>,
+}
 
-        fn flush(&mut self) -> std::io::Result<()> {
-            Ok(())
+/// This is required since we originally supported `Warning` and uppercase variants being used as
+/// the log level filter. It would be a breaking change to no longer support this. In the next
+/// breaking release this should be removed (replaced with `log::LevelFilter` and only supporting
+/// its default deserialization).
```

Comment:
Can be marked as TODO

Reply:
Yes you are right, I will create an issue for this before the PR merges but when it has 2 approved reviews.